students/vkis/lesson09/mailroom5.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+"""
+Lession 09 - Mailroom v5
+=================================================================
+The class Donors is user base (like-private), Donor.name only has access to it's donations (not other users)
+Example: Donor("Vik").donations
+ adding: Donor("Vik").donations = 500
+
+Accessing all user data (outputs in list)
+Data().allnames()
+Data().alldonations()
+Data().allgifts()
+"""
+
+class Data:
+    data_dict = {}
+
+    def __init__(self, name=""):
+        # current name
+        self.name = name
+        # add to dictionary
+        if name != "":
+            try:
+                self.data_dict[name]
+            except KeyError:
+                self.data_dict.update({name: [0, 0]})
+
+    def donations(self, val=0):
+        print("data donations")
+        # return current donation amount for name
+        c_don = self.data_dict[self.name][0]
+        c_gift = self.data_dict[self.name][1]
+        if val == 0: add_gift = 0
+        else: add_gift = 1
+        self.data_dict.update({self.name: [val + c_don, c_gift + add_gift]})
+        return self.data_dict[self.name][0]
+
+    def gifts(self, val=None):
+        if val is not None:
+            c_don = self.data_dict[self.name][0]
+            self.data_dict.update({self.name: [c_don, val]})
+        # return current donation amount for name
+        return self.data_dict[self.name][1]
+
+    @staticmethod
+    def allnames():
+        return [i for i in iter(Data.data_dict)]
+
+    @staticmethod
+    def alldonations():
+        return [Data.data_dict[i][0] for i in iter(Data.data_dict)]
+
+    @staticmethod
+    def allgifts():
+        return [Data.data_dict[i][1] for i in iter(Data.data_dict)]
+
+    @staticmethod
+    def avgDpG():
+        return [Data.data_dict[i][0]/Data.data_dict[i][1] for i in iter(Data.data_dict)]
```

Comment:
With my comment on line 17 done, all the functions from line 46 to 59 should be non static.  And you should reference the data_dict via `self.data_dict`  Static method declarations are for other files or classes or functions to use them OUTSIDE of the current class.  Not for interior of the class itself.